src/incentives/ccip_bridge.sol

```diff
@@ -0,0 +1,190 @@
+import {IRouterClient} from "@chainlink/contracts-ccip/src/v0.8/ccip/interfaces/IRouterClient.sol";
+
+interface IAffinePass {
+    function bridgeMint(address to, uint256 tokenId) external;
+    function bridgeBurn(uint256 tokenId) external;
+    function ownerOf(uint256 tokenId) external view returns (address);
+}
+
+
+contract AffineBridge is CCIPReceiver, Ownable {
+    IAffinePass public affinePass;
+    mapping(uint64 => bool) public whitelistedDestinationChains;
+    mapping(uint64 => bool) public whitelistedSourceChains;
+    mapping(address => bool) public whitelistedSenders;
+    mapping(uint64 => address) public chainReciever;
+    bool public paused = true;
+
+    error NotEnoughBalance(uint256 currentBalance, uint256 calculatedFees); // Used to make sure contract has enough balance.
+    error DestinationChainNotWhitelisted(uint64 destinationChainSelector); // Used when the destination chain has not been whitelisted by the contract owner.
+    error SourceChainNotWhitelisted(uint64 sourceChainSelector); // Used when the source chain has not been whitelisted by the contract owner.
+    error SenderNotWhitelisted(address sender); // Used when the sender has not been whitelisted by the contract owner.
+
+    // Event emitted when a message is sent to another chain.
+    event BridgeRequest(
+        bytes32 indexed messageId,
+        uint64 indexed destinationChainSelector,
+        address sender,
+        uint256 id 
+    );
+
+    // Event emitted when a message is received from another chain.
+    event BridgeReciept(
+        bytes32 indexed messageId,
+        uint64 indexed sourceChainSelector,
+        address receiver,
+        uint256 id
+    );
+
+    modifier onlyWhitelistedDestinationChain(uint64 _destinationChainSelector) {
+        if (!whitelistedDestinationChains[_destinationChainSelector])
+            revert DestinationChainNotWhitelisted(_destinationChainSelector);
+        _;
+    }
+
+    modifier onlyWhitelistedSourceChain(uint64 _sourceChainSelector) {
+        if (!whitelistedSourceChains[_sourceChainSelector])
+            revert SourceChainNotWhitelisted(_sourceChainSelector);
+        _;
+    }
+
+    modifier onlyWhitelistedSenders(address _sender) {
+        if (!whitelistedSenders[_sender]) revert SenderNotWhitelisted(_sender);
+        _;
+    }
+
+    modifier canBridge() {
+        require(!paused, "Bridging is paused");
+        _;
+    }
+
+    constructor(address router) CCIPReceiver(router) {
+        // Polygon
+        whitelistedDestinationChains[4051577828743386545] = true;
+        whitelistedSourceChains[4051577828743386545] = true;
+
+        // ETH
+        whitelistedDestinationChains[5009297550715157269] = true;
+        whitelistedSourceChains[5009297550715157269] = true;
+    }
+
+    fallback() external payable { }
+    receive() external payable { }
+
+    function setAffinePassAddress(address _affinePass) external onlyOwner {
```

Comment:
Is this address supposed to change? Can probably just set this in the constructor as an immutable.